hive/src/main/java/org/apache/iceberg/hive/ClientPool.java

```diff
@@ -133,7 +131,9 @@ public void close() {
 
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
-      LOG.warn("Interrupted while shutting down pool. Some clients may not be closed.");
+      LOG.warn("Interrupted while shutting down pool. Some clients may not be closed.", e);
     }
   }
+
+  protected abstract void close(C client);
```

Comment:
This one might be questionable. Our code style requires overloaded methods to be together.

Reply:
Can we move the other close to just below the abstract methods instead?

Reply:
Done